snyk/snyk
# feat: Delete local .iac-data directory after scan is finished [CC-765]

#### What does this PR do?

In a previous task, we introduced the download and extraction of files in a local cache directory, called '.iac-data'. 

We do not need the files anymore after the scan/test, so we decided to clean up that local directory after the tests finishes.

We also move the dev dependency `rimraf` from dev deps to prod deps. When we support Node >= v12.10.0 we can switch to the supported rmdirSync with recursive options.
https://snyk.slack.com/archives/C0127HWU0E7/p1616688051070900

#### How should this be manually tested?
- Run `npm run build && node iac test file.tf --experimental`
- Run ` ls .iac-data` and see that there is no such directory in the filesystem anymore.

#### Any background context you want to provide?
The package [rimraf](https://www.npmjs.com/package/rimraf) was used for the deletion of the directory. This was already in our dev dependencies. 

_Option 2:_
In our version of node fs.rmdirSync is not supported, so in order to delete a non empty directory we would need to make our own recursive function, to check for files in the directory, unlink them and then delete them one by one. Rimraf does this in one line. That function would look like this:

```
export const cleanLocalCache = (directoryPath) => {
  if (fs.existsSync(directoryPath)) {
    fs.readdirSync(directoryPath).forEach((file) => {
      const curPath = path.join(directoryPath, file);
      fs.unlinkSync(curPath);
    });
    fs.rmdirSync(directoryPath);
  } else {
    console.log('Directory path not found');
  }
};
```

#### What are the relevant tickets?
[CC-765]

[CC-765]: https://snyksec.atlassian.net/browse/CC-765
